app/bundles/CoreBundle/Test/MauticMysqlTestCase.php

```diff
@@ -266,19 +268,20 @@ private function generateResetDatabaseSql(string $file): void
      */
     private function dumpToFile(string $sqlDumpFile): void
     {
-        $connection = $this->connection;
-        $command    = 'mysqldump --opt -h"${:db_host}" -P"${:db_port}" -u"${:db_user}" "${:db_name}" > "${:db_backup_file}"';
-        $envVars    = [
-            'MYSQL_PWD'      => $this->connection->getParams()['password'],
-            'db_host'        => $this->connection->getParams()['host'],
-            'db_port'        => $this->connection->getParams()['port'],
-            'db_user'        => $this->connection->getParams()['user'],
-            'db_name'        => $this->connection->getParams()['dbname'],
-            'db_backup_file' => $sqlDumpFile,
-        ];
+        $connectionParams = $this->connection->getParams();
+        $password         = $connectionParams['password'] ? '-p'.escapeshellarg($connectionParams['password']) : '';
+        $command          = sprintf(
+            'mysqldump --opt -h%s -P%s -u%s %s %s > %s',
```

Comment:
```suggestion
            'mysqldump --opt --host=%s --port=%s --user=%s --password=%s %s > %s',
```
To make it more readable and clear.